src/test/java/net/openhft/compiler/CompilerTest.java

```diff
@@ -286,6 +417,19 @@ public void test_compilerErrorsDoNotBreakNextCompilations() throws Exception {
         assertEquals("ok", callable.call());
     }
 
+    public void test_compilerErrorsButLoadingDifferentClass() throws Exception {
+        // quieten the compiler output
+        PrintWriter quietWriter = new PrintWriter(new StringWriter());
+
+        // TODO: Should this throw an exception due to the compilation error nonetheless to be less
+        // error-prone for users, even if loading class would succeed?
+        Class<?> testClass = compiler.loadFromJava(classLoader,
+                // Load other class which is unaffected by compilation error
+                String.class.getName(),
+                "clazz X {}", quietWriter);
+        assertSame(String.class, testClass);
+    }
```

Comment:
Open question: This mirrors the existing behavior, but personally I think it would be cleaner to always have `CachedCompiler` throw an exception (`ClassNotFoundException`?) when compilation errors occur. Even if loading the class would succeed in the end.